LibVLCSharp/Shared/LibVLC.cs

```diff
@@ -623,28 +633,27 @@ static void UpdateProgress(IntPtr data, IntPtr dialogId, float position, string
         [MonoPInvokeCallback(typeof(LogCallback))]
         static void OnLogInternal(IntPtr data, LogLevel level, IntPtr ctx, IntPtr format, IntPtr args)
         {
-            if (_log == null) return;
+            if (data == IntPtr.Zero)
+                return;
+
+            var gch = GCHandle.FromIntPtr(data);
+
+            if (!gch.IsAllocated || !(gch.Target is EventHandler<LogEventArgs> logger))
+                return;
 
-            IntPtr buffer = IntPtr.Zero;
             try
             {
-                buffer = Marshal.AllocHGlobal(2048 + 1);
-                var count = MarshalUtils.vsprintf(buffer, format, args);
-                if (count > -1)
-                {
-                    var message = buffer.FromUtf8();
-                    GetLogContext(ctx, out var module, out var file, out var line);
+                var message = MarshalUtils.GetLogMessage(format, args);
+
+                GetLogContext(ctx, out var module, out var file, out var line);
 #if NET40
-                    Task.Factory.StartNew(() => _log?.Invoke(null, new LogEventArgs(level, message, module, file, line)));
+                Task.Factory.StartNew(() => logger?.Invoke(null, new LogEventArgs(level, message, module, file, line)));
```

Comment:
shall we lock the calls to the logger? (more a self reminder than a real comment)

Reply:
Why? The null-conditional check is thread-safe.

Reply:
Is the invocation atomic?
I see a risk of messing things up /deadlocking if someone unregister in the log callback. We should think carefully about this I guess.

Reply:
> someone unregister in the log callback. 

Yeah well, that's beyond asking for trouble lool

Reply:
Don't underestimate what users can do 😃. My point is that : What if you unregister a log callback while it is being invoked, say because your log callback takes some time to execute and that you unfortunately unregister the callback at the same time?

My rule of thumb is that if you lock() somewhere, then you should lock everywhere it's being used

Reply:
What you're saying would apply for all events and so far no bug report about this. And a lock in this case would really slow things down as it can be called quite a lot. 

I say let's leave it as it is for now, and see.